core/src/main/java/io/paradoxical/cassieq/model/time/ExecutionServiceScheduler.java

```diff
@@ -0,0 +1,24 @@
+package io.paradoxical.cassieq.model.time;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+public class ExecutionServiceScheduler implements Scheduler {
+    final ScheduledExecutorService scheduledExecutorService;
+
+    public ExecutionServiceScheduler(int periodicThreadPoolSize) {
+        scheduledExecutorService = Executors.newScheduledThreadPool(periodicThreadPoolSize);
+    }
+
+    @Override
+    public ScheduledFuture<?> periodicWithDelay(final Runnable runnable, final long delay, final long duration, final TimeUnit unit) {
```

Comment:
since we own these contracts, delay and duration could be a `Duration` type right?